docs/Hooks.md

```diff
@@ -284,6 +284,23 @@ fastify.addHook('onRoute', (routeOptions) => {
   routeOptions.prefix
 })
 ```
+
+If you are authoring a plugin and you need to customize application routes, like modifying the options or adding new route hooks, this is the right place.
+
+```js
+fastify.addHook('onRoute', (routeOptions) => {
+  // Can be undefined or an array of functions.
+  if(!routeOptions.preSerialization) {
```

Comment:
```js
routeOptions.preSerialization = [...routeOptions.preSerialization, onPreSerialization]

function onPreSerialization (request, reply, payload, done) { ... }
```
```

Reply:
I don't think if it is a best approach for add preSerialization hook inside hook.

Reply:
I can (and will) change the code in few minutes.
Can you please clarify your last comment?

Reply:
Code updated again.